### PR TITLE
Proposal neurons redesign

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

Redesign proposal detail neurons block (split in 3 groups and render in collapsibles). [design](https://www.figma.com/file/29OdZHbNaOk9RkrfESucjb/NNS-1.03-WIP?node-id=5091%3A8121&mode=dev)

# Changes

- update MyVotes container and it's children to render in collapsibles.
- labels updated.
- render user votes with color.
- refactor: VotingCard to be shared between nns and sns versions.

# Tests

- Updated current tests to the new DOM changes.
- groups visibility tests when no (votable, voted, ineligible)neurons.
- voted neurons in color.
- refactor to cleanup a bit the structure:
```
  NnsVotingCard|NnsVotingCard
    VotingCard
      VotableNeuronList
      VotedNeuronList
      IneligibleNeuronList
```
# Todos

- [x] Add entry to changelog (if necessary).

# Screenshots

Screenshot 2024-01-03 at 09 43 01

image

image

### CODE REVIEW

File: frontend/src/lib/components/proposal-detail/VotingCard/ExpandableProposalNeurons.svelte [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    padding-bottom: var(--padding-2x);
+    border-bottom: 1px solid var(--tertiary);
+  }
+
+  .header {
+    display: flex;
+    justify-content: space-between;
+    gap: var(--padding);
+    width: 100%;
+    //margin-bottom: var(--padding-3x);
```
Comment (peterpeterparker) [inline frontend/src/lib/components/proposal-detail/VotingCard/ExpandableProposalNeurons.svelte:55]: leftover?

File: frontend/src/lib/components/proposal-detail/VotingCard/VotingCard.svelte [hunk 2]
```diff
@@ -165,14 +176,37 @@
...
 
-  .container:not(.signedIn) {
+  // TODO(max): combine w/ sns version?
+  .container {
     display: flex;
-    justify-content: center;
-    padding: var(--padding-2x) 0;
+
+    // add scrollbars for too long content
+    max-height: 100vh;
```
Comment (peterpeterparker) [inline frontend/src/lib/components/proposal-detail/VotingCard/VotingCard.svelte:184]: works on iOS?

File: frontend/src/lib/components/proposal-detail/VotingCard/VotingCardNeuronList.svelte [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+    display: flex;
+    flex-direction: column;
+    gap: var(--padding-2x);
+    margin-top: var(--padding);
+
+    --checkbox-padding: 0;
+
+    // fix unexpected scrollbars in voted and ineligible list
+    line-height: normal;
```
Comment (peterpeterparker) [inline frontend/src/lib/components/proposal-detail/VotingCard/VotingCardNeuronList.svelte:16]: this comes few times no in the code, can we solve it more generally?

File: frontend/src/lib/components/proposal-detail/VotingCard/VotingNeuronSelect.svelte [hunk 4]
```diff
@@ -17,96 +18,67 @@
...
-      </button>
-    </div>
-
-    {#if displayNeuronsInfo}
-      <div
-        class="total-voting-power"
-        data-tid="voting-collapsible-toolbar-voting-power"
-      >
+<div class="container" data-tid="voting-neuron-select">
+  {#if totalVotingNeurons > 0}
```
Comment (peterpeterparker) [inline frontend/src/lib/components/proposal-detail/VotingCard/VotingNeuronSelect.svelte:29]: it's a value that can change after user have voted right, would it be interesting to fade it's content apparence?

Comment (peterpeterparker) [reply frontend/src/lib/components/proposal-detail/VotingCard/VotingNeuronSelect.svelte:29]: likewise for other `> 0` checks of this component here under

File: frontend/src/lib/components/proposal-detail/VotingCard/VotingNeuronSelect.svelte [hunk 5]
```diff
@@ -17,96 +18,67 @@
...
-  <VotingNeuronSelectContainer>
-    <slot />
-  </VotingNeuronSelectContainer>
-</Collapsible>
+      </svelte:fragment>
+      <slot />
+    </ExpandableProposalNeurons>
+  {/if}
+
+  {#if votedNeuronCount > 0}
```
Comment (peterpeterparker) [inline frontend/src/lib/components/proposal-detail/VotingCard/VotingNeuronSelect.svelte:49]: Would it be interesting to have three dedicated small components for those totalVotingNeurons, votedNeuronCound and ineligibleNeuronCount?

File: frontend/src/lib/components/sns-proposals/SnsVotingCard.svelte [hunk 6]
```diff
@@ -140,6 +140,13 @@
     });
   }
 
+  let votedVotingPower: bigint;
+  $: votedVotingPower =
```
Comment (peterpeterparker) [inline frontend/src/lib/components/sns-proposals/SnsVotingCard.svelte:144]: this appears for the second time in this PR, worth a utils function?

Comment (mstrasinskis) [reply frontend/src/lib/components/sns-proposals/SnsVotingCard.svelte]: Good point, `neuronsVotingPower` was added.

File: frontend/src/lib/components/proposal-detail/VotingCard/VotingNeuronSelect.svelte (hunk 4 above)
Comment (mstrasinskis) [reply frontend/src/lib/components/proposal-detail/VotingCard/VotingNeuronSelect.svelte:29]: Nice idea! After playing a bit with animation, `in:fade` was added.

File: frontend/src/lib/components/proposal-detail/VotingCard/VotingCardNeuronList.svelte (hunk 3 above)
Comment (mstrasinskis) [reply frontend/src/lib/components/proposal-detail/VotingCard/VotingCardNeuronList.svelte:16]: The only way I found is to replace the font. Let's discuss this w/ designer when he's back.
Screen Recording 2024-01-03 at 15 59 24

File: frontend/src/lib/components/proposal-detail/VotingCard/VotingNeuronSelect.svelte (hunk 5 above)
Comment (mstrasinskis) [reply frontend/src/lib/components/proposal-detail/VotingCard/VotingNeuronSelect.svelte]: Good idea, the VotingNeuronSelect was divided into three and removed:
- votable neurons https://github.com/dfinity/nns-dapp/pull/4120/files#diff-0259d6acb152589f8f04d45cf0d1da0ffa43cc7ed9ea351d2772735913d61271
- voted neurons https://github.com/dfinity/nns-dapp/pull/4120/files#diff-6623baf70d115f3c7bc62b7bfe18e72ac3bee8a6ba4876582c38d7a22aac8f6d
- ineligible neurons https://github.com/dfinity/nns-dapp/pull/4120/files#diff-1f28b53a839c6d73f89af862ad39e8d00fdd9fd593a71cb6604037c18ed851f8

File: frontend/src/lib/components/proposal-detail/VotingCard/ExpandableProposalNeurons.svelte (hunk 1 above)
Comment (mstrasinskis) [reply frontend/src/lib/components/proposal-detail/VotingCard/ExpandableProposalNeurons.svelte]: Thanks, removed.

File: frontend/src/lib/components/proposal-detail/VotingCard/NnsVotingCard.svelte [hunk 7]
```diff
@@ -0,0 +1,133 @@
+<script lang="ts">
```
Comment (mstrasinskis) [inline frontend/src/lib/components/proposal-detail/VotingCard/NnsVotingCard.svelte:1]: @peterpeterparker, before the `<script>` part of this file was in `VotingCard.svelte`. Now the `VotingCard` is agnostic to universes.

File: frontend/src/lib/components/proposal-detail/VotingCard/VotableNeuronList.svelte [hunk 8]
```diff
@@ -0,0 +1,47 @@
+<script lang="ts">
```
Comment (mstrasinskis) [inline frontend/src/lib/components/proposal-detail/VotingCard/VotableNeuronList.svelte:1]: Moved from VotingNeuronSelect to have 3 separate neuron lists (votable, voted, ineligible)

**Comment by mstrasinskis** [comment]
> Regarding design:
> 
> * In the Figma file there is `:` after `Voting power` and it's capitalized. i.e `Voting Power:`
> * Likewise "4 neurons voted" is also capitalized in the design but, not sure we want to change that?

Were fixed!

> * The mobile view looks a bit weird to me, what do you think, would it be worth displaying information under one column instead of two?

@peterpeterparker, thank you for suggestions. Let's collect opinions and improve it in a separate pr.

File: frontend/src/lib/components/proposal-detail/VotingCard/VotingNeuronSelect.svelte (hunk 5 above)
Comment (peterpeterparker) [reply frontend/src/lib/components/proposal-detail/VotingCard/VotingNeuronSelect.svelte]: Cool

File: frontend/src/lib/components/proposal-detail/VotingCard/VotingNeuronSelect.svelte (hunk 4 above)
Comment (peterpeterparker) [reply frontend/src/lib/components/proposal-detail/VotingCard/VotingNeuronSelect.svelte]: Nice !

**Review by peterpeterparker — APPROVED** [review]
LGTM, thx

File: frontend/src/lib/components/proposal-detail/VotingCard/VotingCard.svelte (hunk 2 above)
Comment (mstrasinskis) [reply frontend/src/lib/components/proposal-detail/VotingCard/VotingCard.svelte]: Great catch! A dirty fix was added https://github.com/dfinity/nns-dapp/pull/4120/files#diff-2ec9e42d86e25867e861a03268f0bf8ce159201e306b4351cf618cbe510a0398R67
Screen Recording 2024-01-05 at 12 39 07

**Pull request merged**